frontend/class-public.php

```diff
@@ -308,20 +325,40 @@ public function hide_default_read_more( $more_text ){
 	 *
 	 * Displays the volunteer opportunities lists for both the one-time and flexible
 	 * opportunities. It also calls template files to output the majority of the HTML.
-	 * 
-	 * @param  string $list_type One-time or flexible volunteer opportunities
-	 * @param  array $query_args The query arguments to be used in WP_Query 
+	 *
+	 * @param  string $list_type One-time or flexible volunteer opportunities.
+	 * @param  array  $query_args The query arguments to be used in WP_Query.
+	 * @param array  $block_attributes  Attributes saved from the block editor to display in the admin or frontend.
 	 * @return string            HTML code to be output via a shortcode.
 	 */
-	public function display_volunteer_opp_list( $list_type, $query_args ){
-		//We must edit the main query in order to handle pagination.
+	public function display_volunteer_opp_list( $list_type, $query_args, $block_attributes ) {
+		// We must edit the main query in order to handle pagination.
 		global $wp_query;
 		$temp = $wp_query;
 		$wp_query = new WP_Query( $query_args );
 
-		ob_start(); ?>
-		
-		<div class="volunteer-opps <?php echo $list_type; ?>">
+		ob_start();
+
+		$class_name = $list_type;
+		$anchor     = '';
+
+		if ( is_array( $block_attributes ) ) {
+
+			// If the Additonal CSS class(es) field on the block is used, include it in the container class name.
+			if ( array_key_exists( 'className', $block_attributes ) && $block_attributes['className'] !== '' ) {
```

Comment:
I wonder if you could replace these multiple conditionals by using `empty()` (https://www.php.net/manual/en/function.empty.php). That may allow you to add the parent `if ( is_array( $block_attributes ) ) {` to each one so you don't have to nest them.

Reply:
Just want to make sure I understand what you're saying here.

Remove the is_array() check and instead add empty( $block_attributes ) to both conditonals that check if a block attribute is set?

The reason I nested them is so we wouldn't have to check for is_array() or empty() more than once. Once we know if it's not empty, we shouldn't need to check again.

Let me know if I'm misunderstanding this.

Reply:
Sorry I wasn't clear on this front. I'm thinking something like this might work:

`if ( ! empty( $block_attributes['className'] ) ) {`

With that you might be able to remove `if ( is_array( $block_attributes ) ) {` and replace the conditionals that look like `if ( array_key_exists( 'className', $block_attributes ) && $block_attributes['className'] !== '' ) {`. You'll have to test it, but I think `empty()` checks if the variable is set and whether the value is truthy.

Feel free to shoot me a chat if you want to talk this over.

Reply:
Ahhh, I see I see! I tried this out and empty( $block_attributes['className'] ) does indeed allow us to remove the is_array check, and it completely replaces the conditionals I had written for each block attribute.

Thanks a ton for linking the empty() documentation - this is will be so much better now.